apps/web/src/components/Composer/NewPublication.tsx

```diff
@@ -691,12 +724,44 @@ const NewPublication: FC<NewPublicationProps> = ({ publication }) => {
         );
       }
 
+      // Create Space in Huddle
+      let spaceId = null;
+      if (
+        showNewPublicationModal &&
+        modalPublicationType === NewPublicationTypes.Spaces
+      ) {
+        spaceId = await createSpace();
+      }
+
+      const now = new Date();
+      now.setHours(Number(spacesTimeInHour));
+      now.setMinutes(Number(spacesTimeInMinute));
+      const userTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+      const formattedTime = new Date(
+        now.toLocaleString('en-US', { timeZone: userTimezone })
+      );
+      const startTime = formattedTime.toISOString();
+
       const attributes: MetadataAttributeInput[] = [
         {
           traitType: 'type',
           displayType: PublicationMetadataDisplayTypes.String,
           value: getMainContentFocus()?.toLowerCase()
         },
+        ...(showNewPublicationModal &&
+        modalPublicationType === NewPublicationTypes.Spaces
+          ? [
+              {
+                traitType: 'audioSpace',
+                displayType: PublicationMetadataDisplayTypes.String,
+                value: JSON.stringify({
+                  id: spaceId,
```

Comment:
what will happen if spaceId is not present? 🤔

Reply:
Added a check for spaceId